core/lib/spree/awesome_nested_set_override.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module AwesomeNestedSetOvveride
+  # Add :polimorphic key option only when used to make it work with Rails 6.1+
+  # https://github.com/osmaelo/rails/commit/2c008d9f6311d92b3660193285230505e74a114f
+  # This can be removed when upgrading to an awesome_nested_set version
+  # compliant with Rails 6.1+.
+  def acts_as_nested_set_relate_parent!
+    # Disable Rubocop to keep original code for diffs
+    # rubocop:disable
+    options = {
+      :class_name => self.base_class.to_s,
+      :foreign_key => parent_column_name,
+      :primary_key => primary_column_name,
+      :counter_cache => acts_as_nested_set_options[:counter_cache],
+      :inverse_of => (:children unless acts_as_nested_set_options[:polymorphic]),
+      :touch => acts_as_nested_set_options[:touch]
+    }
+    options[:polymorphic] = true if acts_as_nested_set_options[:polymorphic]
+    options[:optional] = true if ActiveRecord::VERSION::MAJOR >= 5
+    belongs_to :parent, options
+    # rubocop:disable
+  end
+
+  CollectiveIdea::Acts::NestedSet.prepend self
```

Comment:
I'm not sure about adding this kind of patches in Solidus core. I'm thinking that maybe I prefer to keep the build failing and showing exactly which gems need some changes. 

Maybe we should just propose the patch to https://github.com/collectiveidea/awesome_nested_set?

Reply:
Opening a PR on `awesome_nested_set` is a good idea, I'll do it as soon as possibile 👍 

Leaving the build broken could be a problem, Rails master is needed to work on some other issues and `awesome_nested_set`, as it is, breaks the code during initialization when Taxon class is loaded.  Is there a less intrusive solution we can try?

Reply:
We could create a separate/additional temporary Gemfile for Rails 6.1 only, that points to PRs on GitHub that address the Rails 6.1 issues. But it's not guaranteed that they already exist so maybe we can also create a file that is loaded only with Rails 6.1 and contains all the monkey patch that we have to remove before supporting that specific version.

Reply:
Now the monkey patch is loaded only with Rails 6.1+. When [this nested_awesome_set PR](https://github.com/collectiveidea/awesome_nested_set/pull/421) will be merged and released we can remove it altogether.